pkg/cli/admin/upgrade/status/controlplane.go

```diff
@@ -0,0 +1,78 @@
+package status
+
+import (
+	"io"
+	"text/template"
+
+	v1 "github.com/openshift/api/config/v1"
+)
+
+type assessmentState string
+
+const (
+	assessmentStateProgressing assessmentState = "Progressing"
+	assessmentStateCompleted   assessmentState = "Completed"
+)
+
+type operators struct {
+	Total       int
+	Available   int
+	Progressing int
+	Degraded    int
+}
+
+type controlPlaneStatusDisplayData struct {
+	Assessment assessmentState
+	Completion float64
+	Operators  operators
+}
+
+func assessControlPlaneStatus(cv *v1.ClusterVersion, operators []v1.ClusterOperator) controlPlaneStatusDisplayData {
+	var displayData controlPlaneStatusDisplayData
+	var completed int
+
+	for _, operator := range operators {
+		displayData.Operators.Total++
+		for _, condition := range operator.Status.Conditions {
+			switch {
+			case condition.Type == v1.OperatorAvailable:
+				if condition.Status == v1.ConditionTrue {
+					displayData.Operators.Available++
+				}
+			case condition.Type == v1.OperatorProgressing && condition.Status == v1.ConditionTrue:
+				displayData.Operators.Progressing++
+			case condition.Type == v1.OperatorDegraded && condition.Status == v1.ConditionTrue:
+				displayData.Operators.Degraded++
+			}
+		}
+
+		for _, version := range operator.Status.Versions {
+			if version.Name == "operator" {
+				if version.Version == cv.Spec.DesiredUpdate.Version {
```

Comment:
I think we should use `cv.Status.Desired.Version` here.  The `spec` property may not set a version (e.g. via a `--to-image` update request).  And the `spec` property may not be accepted as a new update target, while the `status` property will (or at least should) only include releases once they've been accepted as a target.

Reply:
Done in current HEAD 12bd58bcaf3a731aff4e54fc13a53278989bd7b0